sdk/eventhubs/azure-messaging-eventhubs/inc/azure/messaging/eventhubs/event_data_batch.hpp

```diff
@@ -116,7 +116,8 @@ namespace Azure { namespace Messaging { namespace EventHubs {
      *
      * @returns true if the message was added to the batch, false otherwise.
      */
-    bool TryAddMessage(std::shared_ptr<Azure::Core::Amqp::Models::AmqpMessage const> const& message)
+    _azure_NODISCARD bool TryAdd(
```

Comment:
is this really needed ?

Reply:
IMHO, this is the canonical example of when NODISCARD is needed. If you don't check for the result of TryAdd, then you run the risk of losing messages, which in a messaging application is a data loss situation.